tests/test_pay.py

```diff
@@ -1873,131 +1873,144 @@ def test_setchannelfee_usage(node_factory, bitcoind):
     node_factory.join_nodes([l1, l2])
     l1.rpc.connect(l3.info['id'], 'localhost', l3.port)
 
-    def channel_get_fees(scid):
+    def channel_get_config(scid):
         return l1.db.query(
-            'SELECT feerate_base, feerate_ppm FROM channels '
+            'SELECT feerate_base, feerate_ppm, htlc_maximum_msat FROM channels '
             'WHERE short_channel_id=\'{}\';'.format(scid))
 
     # get short channel id
     scid = l1.get_channel_scid(l2)
 
     # feerates should be init with global config
-    db_fees = l1.db_query('SELECT feerate_base, feerate_ppm FROM channels;')
+    db_fees = l1.db_query('SELECT feerate_base, feerate_ppm, htlc_maximum_msat FROM channels;')
     assert(db_fees[0]['feerate_base'] == DEF_BASE)
     assert(db_fees[0]['feerate_ppm'] == DEF_PPM)
+    # This will be the capacity - reserves:
+    assert(db_fees[0]['htlc_maximum_msat'] == MAX_HTLC)
     # this is also what listpeers should return
     peers = l1.rpc.listpeers()['peers']
     assert peers[0]['channels'][0]['fee_base_msat'] == DEF_BASE_MSAT
     assert peers[0]['channels'][0]['fee_proportional_millionths'] == DEF_PPM
+    assert peers[0]['channels'][0]['maximum_htlc_out_msat'] == MAX_HTLC
 
-    # custom setchannelfee scid <base> <ppm>
-    result = l1.rpc.setchannelfee(scid, 1337, 137)
+    # custom setchannel scid <feebase> <feeppm> <htlcmax>
+    result = l1.rpc.setchannel(scid, 1337, 137, 133337)
 
     # check result format
-    assert(result['base'] == 1337)
-    assert(result['ppm'] == 137)
     assert(len(result['channels']) == 1)
     assert(re.match('^[0-9a-f]{64}$', result['channels'][0]['channel_id']))
     assert(result['channels'][0]['peer_id'] == l2.info['id'])
     assert(result['channels'][0]['short_channel_id'] == scid)
+    assert(result['channels'][0]['fee_base_msat'] == 1337)
+    assert(result['channels'][0]['fee_proportional_millionths'] == 137)
+    assert(result['channels'][0]['maximum_htlc_out_msat'] == 50000)
 
     # check if custom values made it into the database
-    db_fees = channel_get_fees(scid)
+    db_fees = channel_get_config(scid)
     assert(db_fees[0]['feerate_base'] == 1337)
     assert(db_fees[0]['feerate_ppm'] == 137)
+    assert(db_fees[0]['htlc_maximum_msat'] == 133337)
     # also check for updated values in `listpeers`
     peers = l1.rpc.listpeers()['peers']
     assert peers[0]['channels'][0]['fee_base_msat'] == Millisatoshi(1337)
     assert peers[0]['channels'][0]['fee_proportional_millionths'] == 137
+    assert peers[0]['channels'][0]['maximum_htlc_out_msat'] == 133337
 
     # wait for gossip and check if l1 sees new fees in listchannels
     wait_for(lambda: [c['base_fee_millisatoshi'] for c in l1.rpc.listchannels(scid)['channels']] == [DEF_BASE, 1337])
     wait_for(lambda: [c['fee_per_millionth'] for c in l1.rpc.listchannels(scid)['channels']] == [DEF_PPM, 137])
+    wait_for(lambda: [c['htlc_maximum_msat'] for c in l1.rpc.listchannels(scid)['channels']] == [MAX_HTLC, 133337])
```

Comment:
This also fails. `listchannels` currently spits out `990000000msat` for the other direction. Not sure whats correct here since `listchannels` is a gossip function, maybe it should or should not know about counterparts reserve state.